cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -1382,6 +1386,32 @@ TEST_F(TestAzuriteFileSystem, DeleteDirContentsFailureNonexistent) {
   this->TestDeleteDirContentsFailureNonexistent();
 }
 
+TEST_F(AzuriteFileSystemTest, DeleteFileSuccessHaveFile) {
+void CreateFile(FileSystem* fs, const std::string& path, const std::string& data) {
+  const auto file_name = PreexistingData::RandomFileName() +
+    PreexistingData::RandomFileNameExtension();
+  ASSERT_OK(CreateFile(fs_.get(), file_name, "abc"));
+  arrow::fs::AssertFileInfo(fs_.get(), file_name, FileType::File);
+  ASSERT_OK(fs_->DeleteFile(file_name));
+  arrow::fs::AssertFileInfo(fs_.get(), file_name, FileType::NotFound);
+}
```

Comment:
Not certain that I understand the question.  The idea behind this test was to create a file to delete, then delete it.  Thus the pseudo-code was:
1) Create a filename
2) Create the file and verify that the file creation was successful
3) Verify that the file type was a file (could be removed if considered redundant)
4) Delete the file and verify that the delete attempt did not assert
5) Verify that the file is not found (because it has been deleted)
As for the functions RandomFileName() and RandomFileNameExtension(), I created those to clean the code a bit and placed them with the RandomChars() method.  This is just a choice, to abstract the implementation details into a function.  It doesn't have to be that way.  The content of those two methods could be placed inline at lines 1391 and 1392 and these two methods deleted if that is the preferred manner of going about it.  Open to suggestion/guidance/feedback.

Reply:
BTW, the choice to create the RandomFileName() and RandomFileNameExtension() methods was done in part because I prefer abstractions (generally) and also because there was a similar abstraction for the container name, which is found at line 369, thus:
`  static std::string RandomContainerName(RNG& rng) { return RandomChars(32, rng); }
`
And so I decided to follow suit by creating abstractions for file names, just as there was one for container names.  Again, I'll pull those back inline since that seems to be the preference.

Reply:
Alright, now I understand why kou asked why I defined a function in `TEST_F() {...} `.  The short answer is that this was a lack of deletion error on my part.  I had copied the definition of the CreateFile() method temporarily while I wrote the test to ensure that I got the invocation parms correctly written.  Then I simply forgot to delete the reference.  Apologies.  I have deleted that from the most recent version.

Reply:
Finally, due to the structure of how the RandomChars() function is written within the PreexistingData class, I'll have to leave the random-name functionality that I need within this class.  However, I'll take kou's suggestion and combine the two methods, as well as changing the name to blob.

Reply:
The reason we create random containers is to not have test runs interfere with later runs when the tests crash and containers can't be deleted from the storage account. The blobs inside the containers don't have to be random. Every test case is creating a new random container that you can freely add named blobs to.